File: sections/07_foundation_mixed_contest/problems/e_circular_patrol/solution.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    int tests;
    cin >> tests;
    while (tests--) {
        int n, position;
        string commands;
        cin >> n >> position >> commands;
        --position;
        vector<char> seen(n, false);
        seen[position] = true;
        int distinct = 1;
        for (char command : commands) {
            position = (position + (command == 'R' ? 1 : n - 1)) % n;
            if (!seen[position]) {
                seen[position] = true;
                ++distinct;
            }
        }
        cout << position + 1 << ' ' << distinct << '\n';
    }
}

File: sections/07_foundation_mixed_contest/problems/e_circular_patrol/solution.py
import sys


def main() -> None:
    tokens = sys.stdin.buffer.read().split()
    tests = int(tokens[0])
    index = 1
    answers = []
    for _ in range(tests):
        n = int(tokens[index])
        position = int(tokens[index + 1]) - 1
        commands = tokens[index + 2].decode()
        index += 3
        seen = {position}
        for command in commands:
            position = (position + (1 if command == "R" else -1)) % n
            seen.add(position)
        answers.append(f"{position + 1} {len(seen)}")
    print("\n".join(answers))


if __name__ == "__main__":
    main()
